contributors/design-proposals/storage/volume-topology-scheduling.md

```diff
@@ -276,6 +278,109 @@ StorageClasses for the volume types that need the new binding behavior.
 behavior depending on the StorageClass configuration.  We will mitigate this by
 adding a new PVC event to indicate if binding will follow the new behavior.
 
+#### Add a way to override topology in StorageClass
+Provisioning related topology information is currently specified as elements
+of parameters in StorageClass, which is obscure to users. Here we extend
+StorageClass with a new NodeAffinity field that specifies
+the topology constraints, like we does for PersistentVolume:
+
+```
+type StorageClass struct {
+    ...
+    NodeAffinity *VolumeNodeAffinity
+}
+```
+
+Thus, in predicates, the scheduler would check with the topology affinity
+information along with existing NodeAffinity information from the pod when making scheduling decision.
+
+_TBD: the common NodeSelector could hardly cover use case of `replica-zones` and `replica-zones:auto`,
```

Comment:
I think the use cases of replica-zones is covered.

If you specify no nodeaffinity, then it is the "auto" case.
If you specify some nodeaffinity, then it is one of the "zones" or "replica-zones" case.

Reply:
I didn't know much about  "replica-zones" before, but skimming through code in `gce_pd`, it seems that in the case of "replica-zones", user pass multi zones(currently 2) into storageclass, and when provisioning, the storage system tries to create volumes with the zones in function `createRegionalPD` (not sure about  the details behind it, maybe create a volume in each of the zones for backup or HA issue? )
So could the scene be expressed with nodeaffinity?

Reply:
Yes, to mimic replica-zones, you could pass in the nodeaffinity: "zones IN us-central1-a,us-central1-b".  If you set nothing, then provisioner will randomly choose 2 zones.  If you set only 2 zones, then provisioner will choose those two zones.  If you set > 2 zones, provisioner will randomly select 2 from the set.  If you only set 1, then provisioner will probably return error.

Reply:
But how should we distinguish it from the case of "zones"?
My first reaction on the "In" expression you gave would be "randomly select one from the two"(i.e. for the "zones" case). but not "choose both of the two"

Reply:
Ah, we're going to add a new parameters field to gce pd, to indicate if it is regular pd or repd.  Depending on that, it will know how to interpret the node affinity.

Reply:
Okay, I see you already have an update on the doc, I think I should look into it first.
As I'm ooo today, would look into it tomorrow :)